src/Resource.ts

```diff
@@ -156,17 +146,58 @@ export class Resource extends (BaseResource as any)
         for(const p in params)
         {
             const property = this.property(p);
-            if(property && property.type() == "object")
+            if(property && property.type() === "mixed")
                 params[p] = JSON.parse(params[p]);
+            if(property && property.type() === "number" && params[p] && params[p].toString().length)
+                params[p] = +params[p];
+            if(property && property.type() === "reference" && params[p] && params[p].toString().length){
+                /**
+                 * references cannot be stored as an IDs in typeorm, so in order to mimic this) and
+                 * not fetching reference resource) change this:
+                 * { postId: "1" }
+                 * to that:
+                 * { post: { id: 1 } }
+                 */
+                params[property.column.propertyName] = { id: +params[p] };
+            }
+
         }
         return params;
     }
 
+    private async validateAndSave(instance: BaseEntity) {
+        if (Resource.validate) {
```

Comment:
this is quite a hack - class-validator has to be passed to dev dependency because we are using it on tests. But when we pass it like that in the dev mode there are 2 class-validator packages active. The one from the original project and one from the admin-bro-typeorm, and this doesn't work. The way I handled that is to inject the class-validator via Resource.validate - checkout example in the readme.

This is temporary handling validations in typeorm. Long term would be to add something like BaseValidator in AdminBro, but I don't even think how to handle that right now.